setup.sh

```diff
@@ -0,0 +1,289 @@
+helpstring="Usage: source $BASH_SOURCE [options]
+    Options:
+        -p Path to the venv location
+        -c cmsgemos release version (e.g. X.Y.Z)
+        -d debug information is printed
+        -g gemplotting release version (e.g. X.Y.Z)
+        -G gemplotting dev version (e.g. single integer)
+        -h displays this string
+        -v vfatqc release version (e.g. X.Y.Z)
+        -V vfatqc dev version (e.g. single integer)
+        -w No value following, deletes and recreates the venv from scratch
+
+    The virtualenv found at -p will either be activated or created"
+
+# Detect when we're not being sourced, print a hint and exit
+if [[ "$0" = "$BASH_SOURCE" ]];
+then
+    echo >&2 "ERROR: You must use \"source $0\" to run this script."
+    exit 1
+fi
+
+CMSGEMOS_VERSION=""
+DEBUG=""
+GEMPLOT_VERSION=""
+GEMPLOT_DEV_VERSION=""
+VFATQC_VERSION=""
+VFATQC_DEV_VERSION=""
+VENV_ROOT=""
+WIPE=""
+
+OPTIND=1
+while getopts "c:g:G:v:V:p:wh:d" opts
+do
+    case $opts in
+        c)
+            CMSGEMOS_VERSION="$OPTARG";;
+        d)
+            DEBUG="true";;
+        g)
+            GEMPLOT_VERSION="$OPTARG";;
+        G)
+            GEMPLOT_DEV_VERSION="$OPTARG";;
+        v)
+            VFATQC_VERSION="$OPTARG";;
+        V)
+            VFATQC_DEV_VERSION="$OPTARG";;
+        p)
+            VENV_ROOT="$OPTARG";;
+        w)
+            WIPE=1;;
+        h)
+            echo >&2 "${helpstring}"
+            return 1;;
+        \?)
+            echo >&2 "${helpstring}"
+            return 1;;
+        [?])
+            echo >&2 "${helpstring}"
+            return 1;;
+    esac
+done
+unset OPTIND
```

Comment:
why `unset` this here? It's not being exported and should only be valid inside this execution.

Reply:
Not when the script is being sourced, which is its primary usage